File: year-2021/day-01/solution.py
with open("input-01.txt") as fp:
    nums = [ int(i) for i in fp.readlines() ]
    
    part1 = len([ a for (a, b) in zip(nums, nums[1:]) if b > a ])
    part2 = len([ a for (a, b) in zip(nums, nums[3:]) if b > a ])

    print(f'part1: {part1}, part2: {part2}')

File: year-2021/day-01/solution.rs
fn main() -> anyhow::Result<()> {
    let s: Vec<_> = include_str!("input-01.txt")
        .split('\n')
        .filter_map(|s| s.parse::<i64>().ok())
        .collect();

    let part1 = s.iter().zip(s[1..].iter()).filter(|(a, b)| b > a).count();
    let part2 = s.iter().zip(s[3..].iter()).filter(|(a, b)| b > a).count();

    println!("Part 1: {:?}, Part 2: {:?}", &part1, &part2);
    Ok(())
}